pkg/trustedresources/verify.go

```diff
@@ -41,6 +41,29 @@ const (
 	SignatureAnnotation = "tekton.dev/signature"
 )
 
+const (
+	VerificationSkip = iota
+	VerificationPass
+	VerificationWarn
+	VerificationError
+)
+
+// VerificationResultType indicates different cases of a verification result
+type VerificationResultType int
+
+// VerificationResult contains the type and message about the result of verification
+type VerificationResult struct {
+	// VerificationResultType has 4 types which is corresponding to 4 cases:
+	// 1) VerificationSkip: The verification was skipped. Err is nil in this case.
+	// 2) VerificationPass: The verification passed. Err is nil in this case.
+	// 3) VerificationWarn: A warning is logged. It could be no matching policies and feature flag "trusted-resources-verification-no-match-policy" is warn,
+	//    or only Warn mode verification policies fail.
+	// 4) VerificationError: The verification failed, it could be the signature doesn't match the public key or there are errors during verification.
```

Comment:
This could also happen if there are no matching policies and "no-match-policy" is set to "fail" right?

Reply:
ah yes, I missed that case, will add now